server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsAggregatorTests.java

```diff
@@ -0,0 +1,192 @@
+import org.apache.lucene.util.automaton.RegExp;
+import org.elasticsearch.index.mapper.KeywordFieldMapper;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.NumberFieldMapper;
+import org.elasticsearch.search.aggregations.AggregationExecutionException;
+import org.elasticsearch.search.aggregations.AggregatorTestCase;
+import org.elasticsearch.search.aggregations.support.ValueType;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class StringTermsAggregatorTests extends AggregatorTestCase {
+    private static final String LONG_FIELD = "numeric";
+    private static final String KEYWORD_FIELD = "keyword";
+
+    private static final List<Long> dataset;
+    static {
+        List<Long> d = new ArrayList<>(45);
+        for (int i = 0; i < 10; i++) {
+            for (int j = 0; j < i; j++) {
+                d.add((long) i);
+            }
+        }
+        dataset  = d;
+    }
+
+    public void testMatchNoDocs() throws IOException {
+        testBothCases(new MatchNoDocsQuery(), dataset,
+            aggregation -> aggregation.field(KEYWORD_FIELD),
+            agg -> assertEquals(0, agg.getBuckets().size()), ValueType.STRING
+        );
+        testBothCases(new MatchNoDocsQuery(), dataset,
+            aggregation -> aggregation.field(LONG_FIELD),
+            agg -> assertEquals(0, agg.getBuckets().size()), ValueType.NUMERIC
+        );
+    }
+
+    public void testMatchAllDocs() throws IOException {
+        Query query = new MatchAllDocsQuery();
+
+        testBothCases(query, dataset,
+            aggregation -> aggregation.field(LONG_FIELD),
+            agg -> {
+                assertEquals(9, agg.getBuckets().size());
+                for (int i = 0; i < 9; i++) {
+                    LongTerms.Bucket bucket = (LongTerms.Bucket) agg.getBuckets().get(i);
+                    assertThat(bucket.getKey(), equalTo(9L - i));
+                    assertThat(bucket.getDocCount(), equalTo(9L - i));
+                }
+            }, ValueType.NUMERIC
+        );
+        testBothCases(query, dataset,
+            aggregation -> aggregation.field(KEYWORD_FIELD),
+            agg -> {
+                assertEquals(9, agg.getBuckets().size());
+                for (int i = 0; i < 9; i++) {
+                    StringTerms.Bucket bucket = (StringTerms.Bucket) agg.getBuckets().get(i);
+                    assertThat(bucket.getKey(), equalTo(String.valueOf(9L - i)));
+                    assertThat(bucket.getDocCount(), equalTo(9L - i));
+                }
+            }, ValueType.STRING
+        );
+    }
+
+    public void testBadIncludeExclude() throws IOException {
+        IncludeExclude includeExclude = new IncludeExclude(new RegExp("foo"), null);
+
+        // Bytes fails if the formatter is not RAW.  Note this is still on the long field, just hinted as string
```

Comment:
Comment is misleading, I think - this isn't testing the bytes case?  You can tell by the exception message it expects.  The string case doesn't end with "used to filter numeric fields"